library_generation/test/generate_library_unit_tests.sh

```diff
@@ -15,25 +22,21 @@ extract_folder_name_test() {
   assertEquals "google-cloud-aiplatform-v1-java" "${folder_name}"
 }
 
-get_grpc_version_succeed_with_valid_generator_version_test() {
-  local actual_version
-  actual_version=$(get_grpc_version "2.24.0")
-  rm "gapic-generator-java-pom-parent-2.24.0.pom"
-  assertEquals "1.56.1" "${actual_version}"
-}
-
-get_grpc_version_failed_with_invalid_generator_version_test() {
-  local res=0
-  $(get_grpc_version "1.99.0") || res=$?
-  assertEquals 1 $((res))
+get_grpc_version_fails_with_no_env_var_test() {
```

Comment:
Can we move the newly added shell test to `utilities_unit_tests.py` by using `util.sh_util()` method?

I think python unit tests are better maintained than shell tests. WDYT?

Reply:
I was under the impression that `genreate_library_units` is meant for `utilities.sh` and that `utilities_units.py` was for `utilities.py`. 

Could you please explain how are they `py` unit tests better maintained?

Reply:
> Could you please explain how are the py unit tests better maintained?

The shell script doesn't have (or we don't use) a testing framework so we need to implement our own testing utilities, e.g., assert, test counters, etc, in `test_utilities.sh`. Every time we want to add a test, we need to add the test function to the test list.

All these burden was introduced at the beginning stage of this project because we didn't use python at that time.

Right now we have a python language framework already setup and we should migrate all shell script (with tests) to python, although this is out of scope of this PR.

> I was under the impression that genreate_library_unit_tests.sh was meant for utilities.sh and that utilities_units.py was for utilities.py.

For newly added shell tests, I suggest to wrap the test with `util.sh_util()`.

I didn't notice the test is verifying `genreate_library.sh`. Create `genreate_library_unit_tests.py` and put the test there is better. WDYT?